### ISSUE TITLE
MS Edge: Weird rectangles in icons

### ISSUE THREAD

**Issue by dkter** [vector-im/riot-web#8672]
- **Browser**: Microsoft Edge 44.17763.1.0
- **OS**: Windows 10 1809

### Description

Light theme:
![Textbox icons](https://user-images.githubusercontent.com/6720845/52824406-646b1100-3086-11e9-868c-a924350184b1.png)
![Left sidebar icons](https://user-images.githubusercontent.com/6720845/52824415-6d5be280-3086-11e9-89f6-559bad264d0b.png)
![Jump to bottom icon](https://user-images.githubusercontent.com/6720845/52824424-78167780-3086-11e9-8a74-782b26f14bd0.png)
![Top right icons](https://user-images.githubusercontent.com/6720845/52824442-954b4600-3086-11e9-82a5-cba7e7209797.png)

Dark theme:
![Textbox icons](https://user-images.githubusercontent.com/34610081/52824686-90d35d00-3087-11e9-8557-3d66b159092f.png)
![Left sidebar icons](https://user-images.githubusercontent.com/34610081/52824704-a183d300-3087-11e9-88b7-0faacdd2685e.png)
![Jump to bottom icon](https://user-images.githubusercontent.com/34610081/52824710-ac3e6800-3087-11e9-901b-a97ce033ada0.png)
![Top right icons](https://user-images.githubusercontent.com/34610081/52824734-c11afb80-3087-11e9-893e-b8df7f5457a7.png)

It kinda looks like the SVGs' rightmost and lowermost pixels are being extended to the edge of the element. Loading the SVGs themselves in standalone tabs displays them fine.

### Steps to reproduce

- Open Riot in Microsoft Edge (the icons look fine in Firefox).

Log: not sent (this is a visual bug and likely easy to reproduce, I can send a log if you want me to though)

**Comment by turt2live** [comment]
It's a whole lot worse for me:
![image](https://user-images.githubusercontent.com/1190097/52825624-a726ec00-307a-11e9-8ac9-ae668708e54f.png)
![image](https://user-images.githubusercontent.com/1190097/52825626-aa21dc80-307a-11e9-9fb9-f5d30a771476.png)

**Comment by dkter** [comment]
@turt2live Huh. Is that Edge as well?

**Comment by turt2live** [comment]
yup.

**Comment by 2Belette** [comment]
Exactly the same issue on my side.
My browser is Iceweasel-UXP (based on Firefox 52.9)

**Comment by lampholder** [comment]
On reflection I'm going to close this issue - we don't officially support Edge (or Icecat/weasel) so this is more noise than useful.

**Comment by lampholder** [comment]
Good job I didn't close this - the status of Edge is actually "not actively testing but should work". So I guess we do want to fix this.

**Comment by lampholder** [comment]
OKAY the actual real status of Edge is that we do not support it. Good to have a conclusion on that, at least.

So I am going to close this issue. 

There are legitimate arguments for keeping this issue open. Somebody from the community might want to submit a patch to make it work properly on Edge. We might receive a question about how much it would cost to implement Edge support tomorrow, in which case this issue could help inform the size of the delta.

I want to close it because I want to discourage our using github to track issues on unsupported platforms. The set of information on how Riot Web doesn't work on unsupported platforms is impossibly large, so any data we gather is likely to go stale before it is useful. The larger and staler the set of issues we track on github, the less useful that set of issues becomes.

tl;dr the value of knowing how Riot doesn't work on Edge, a platform we don't support < the value of having a set of issues small enough to reason about on a daily basis.

**Comment by 2Belette** [comment]
I agree for Edge but what about the same issue on Iceweasel (based on Firefox 52.9). As Iceweasel is used on Debian and Debian like Linux flavor, perhaps this would deserve to keep a look on it ?

**Comment by jryans** [comment]
Our current support commitment is latest versions of Chrome, Firefox, and Safari. Firefox 52 was released on 2017-03-07, and that is now quite old in the world of browsers. As we are a small team, we need to keep the maintenance burden low by sticking to latest versions where possible.

**Comment by 2Belette** [comment]
It makes perfect sense for make a focus on some browsers, for the Firefox 52, this is because it is an LTS version still maintain and used on LTS distros.

Would it be an option to help you troubleshoot this ? Perhaps we are facing a very easy one to fix

**Comment by jryans** [comment]
The images use the CSS property `mask-image`, so your browser would need to support this property for them to render correctly. It was added to Firefox in version 53.

The current release of Firefox is 67. The current extended support release is 60. If someone wants to contribute a PR to support these older browsers, we could consider it, but I can't make any guarantees unfortunately. If someone wants to work on that, please ask about it in #riot-dev:matrix.org.

**Comment by dkter** [comment]
Edge's issue appears to be the `mask-repeat` property, which is supposed to be supported (according to [Microsoft](https://docs.microsoft.com/en-us/microsoft-edge/dev-guide#css-updates)) but is bugged even in the [MDN demo](https://developer.mozilla.org/en-US/docs/Web/CSS/mask-repeat#Single_value):

![image](https://user-images.githubusercontent.com/34610081/54570658-b967b300-49b5-11e9-8916-a07e208782f0.png)

**Comment by 2Belette** [comment]
Just a quick look on my homeserver logs and I got theses kind or error every minute or less, it is like it is related to my thumbnails issue but not 100% sure, I also note that a lot of time I have to `synctl restart` as my synapse appears disconnected on riot..

```
2019-03-19 09:48:14,017 - synapse.access.http.8008 - 302 - INFO - GET-154 - 10.20.110.1 - 8008 - {None} Processed request: 0.706sec/0.000sec (0.230sec, 0.007sec) (0.000sec/0.000sec/3) 55B 500 "GET /_matrix/media/v1/thumbnail/matrix.fdn.fr/VXVhSTfqPOdQNLwGiQHyJWDa?width=30&height=30&method=crop HTTP/1.1" "Mozilla/5.0 (Windows NT 10.0; WOW64; rv:60.0) Gecko/20100101 Firefox/60.0" [0 dbevts]
2019-03-19 09:48:14,025 - synapse.rest.media.v1.media_repository - 534 - INFO - GET-155 - Stored thumbnail in file u'/home/belette/.synapse/media_store/remote_thumbnail/matrix.fdn.fr/VX/Vh/STfqPOdQNLwGiQHyJWDa/14-14-image-png'
2019-03-19 09:48:14,027 - synapse.http.server - 112 - ERROR - GET-155 - Failed handle request via 'ThumbnailResource': <XForwardedForRequest at 0x3ed028565a8 method=u'GET' uri=u'/_matrix/media/v1/thumbnail/matrix.fdn.fr/VXVhSTfqPOdQNLwGiQHyJWDa?width=14&height=14&method=crop' clientproto=u'HTTP/1.1' site=8008>
Traceback (most recent call last):
  File "/home/belette/.synapse/lib/python2.7/site-packages/synapse/http/server.py", line 81, in wrapped_request_handler
    yield h(self, request)
  File "/home/belette/.synapse/lib/python2.7/site-packages/twisted/internet/defer.py", line 1416, in _inlineCallbacks
    result = result.throwExceptionIntoGenerator(g)
  File "/home/belette/.synapse/lib/python2.7/site-packages/twisted/python/failure.py", line 491, in throwExceptionIntoGenerator
    return g.throw(self.type, self.value, self.tb)
  File "/home/belette/.synapse/lib/python2.7/site-packages/synapse/rest/media/v1/thumbnail_resource.py", line 78, in _async_render_GET
    width, height, method, m_type
  File "/home/belette/.synapse/lib/python2.7/site-packages/twisted/internet/defer.py", line 1416, in _inlineCallbacks
    result = result.throwExceptionIntoGenerator(g)
  File "/home/belette/.synapse/lib/python2.7/site-packages/twisted/python/failure.py", line 491, in throwExceptionIntoGenerator
    return g.throw(self.type, self.value, self.tb)
  File "/home/belette/.synapse/lib/python2.7/site-packages/synapse/rest/media/v1/thumbnail_resource.py", line 221, in _select_or_generate_remote_thumbnail
    desired_height, desired_method, desired_type
  File "/home/belette/.synapse/lib/python2.7/site-packages/twisted/internet/defer.py", line 1416, in _inlineCallbacks
    result = result.throwExceptionIntoGenerator(g)
  File "/home/belette/.synapse/lib/python2.7/site-packages/twisted/python/failure.py", line 491, in throwExceptionIntoGenerator
    return g.throw(self.type, self.value, self.tb)
  File "/home/belette/.synapse/lib/python2.7/site-packages/synapse/rest/media/v1/media_repository.py", line 540, in generate_remote_exact_thumbnail
    t_width, t_height, t_type, t_method, t_len
  File "/home/belette/.synapse/lib/python2.7/site-packages/twisted/internet/defer.py", line 1416, in _inlineCallbacks
    result = result.throwExceptionIntoGenerator(g)
  File "/home/belette/.synapse/lib/python2.7/site-packages/twisted/python/failure.py", line 491, in throwExceptionIntoGenerator
    return g.throw(self.type, self.value, self.tb)
  File "/home/belette/.synapse/lib/python2.7/site-packages/synapse/storage/_base.py", line 516, in _simple_insert
    self._simple_insert_txn, table, values,
  File "/home/belette/.synapse/lib/python2.7/site-packages/twisted/internet/defer.py", line 1416, in _inlineCallbacks
    result = result.throwExceptionIntoGenerator(g)
  File "/home/belette/.synapse/lib/python2.7/site-packages/twisted/python/failure.py", line 491, in throwExceptionIntoGenerator
    return g.throw(self.type, self.value, self.tb)
  File "/home/belette/.synapse/lib/python2.7/site-packages/synapse/storage/_base.py", line 408, in runInteraction
    *args, **kwargs
  File "/home/belette/.synapse/lib/python2.7/site-packages/twisted/internet/defer.py", line 1416, in _inlineCallbacks
    result = result.throwExceptionIntoGenerator(g)
  File "/home/belette/.synapse/lib/python2.7/site-packages/twisted/python/failure.py", line 491, in throwExceptionIntoGenerator
    return g.throw(self.type, self.value, self.tb)
  File "/home/belette/.synapse/lib/python2.7/site-packages/synapse/storage/_base.py", line 457, in runWithConnection
    inner_func, *args, **kwargs
  File "/home/belette/.synapse/lib/python2.7/site-packages/twisted/python/threadpool.py", line 250, in inContext
    result = inContext.theWork()
  File "/home/belette/.synapse/lib/python2.7/site-packages/twisted/python/threadpool.py", line 266, in <lambda>
    inContext.theWork = lambda: context.call(ctx, func, *args, **kw)
  File "/home/belette/.synapse/lib/python2.7/site-packages/twisted/python/context.py", line 122, in callWithContext
    return self.currentContext().callWithContext(ctx, func, *args, **kw)
  File "/home/belette/.synapse/lib/python2.7/site-packages/twisted/python/context.py", line 85, in callWithContext
    return func(*args,**kw)
  File "/home/belette/.synapse/lib/python2.7/site-packages/twisted/enterprise/adbapi.py", line 306, in _runWithConnection
    compat.reraise(excValue, excTraceback)
  File "/home/belette/.synapse/lib/python2.7/site-packages/twisted/enterprise/adbapi.py", line 297, in _runWithConnection
    result = func(conn, *args, **kw)
  File "/home/belette/.synapse/lib/python2.7/site-packages/synapse/storage/_base.py", line 453, in inner_func
    return func(conn, *args, **kwargs)
  File "/home/belette/.synapse/lib/python2.7/site-packages/synapse/storage/_base.py", line 329, in _new_transaction
    r = func(txn, *args, **kwargs)
  File "/home/belette/.synapse/lib/python2.7/site-packages/synapse/storage/_base.py", line 536, in _simple_insert_txn
    txn.execute(sql, vals)
  File "/home/belette/.synapse/lib/python2.7/site-packages/synapse/storage/_base.py", line 119, in execute
    self._do_execute(self.txn.execute, sql, *args)
  File "/home/belette/.synapse/lib/python2.7/site-packages/synapse/storage/_base.py", line 149, in _do_execute
    sql, *args
IntegrityError: UNIQUE constraint failed

```

**Comment by jryans** [comment]
The above log has nothing to do these icons. It's also a Synapse issue, so I would suggest reporting at https://github.com/matrix-org/synapse.

**Comment by 2Belette** [comment]
Iceweasel-UXP updated yesterday to 52.9.20190308 fixed the issue for me!
